SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift

```diff
@@ -1,168 +1,139 @@
-    let storageMaxSize = UserDefaults.standard.object(forKey: kMSStorageMaxSizeKey) as? Int ?? kMSDefaultDatabaseSize
-    self.storageMaxSizeField?.stringValue = "\(storageMaxSize / 1024)"
-  }
-
-  override func controlTextDidChange(_ obj: Notification) {
-    let text = obj.object as? NSTextField
-    if text == self.storageMaxSizeField{
-        let maxSize = Int(self.storageMaxSizeField.stringValue) ?? 0
-        UserDefaults.standard.set(maxSize * 1024, forKey: kMSStorageMaxSizeKey)
-    }
-  }
-
-  // Authentication
-  func showSignInController(action: AuthenticationViewController.AuthAction) {
-    currentAction = action
-    self.performSegue(withIdentifier: "ShowSignIn", sender: self)
-  }
-
-  @IBAction func signInClicked(_ sender: NSButton) {
-    showSignInController(action: AuthenticationViewController.AuthAction.signin)
-  }
-
-  @IBAction func singOutClicked(_ sender: NSButton) {
-    showSignInController(action: AuthenticationViewController.AuthAction.signout)
-  }
-
-  override func prepare(for segue: NSStoryboardSegue, sender: Any?) {
-    if let signInController = segue.destinationController as? AuthenticationViewController {
-        signInController.action = currentAction
-    }
-  }
-}
+import Cocoa
+
+// 10 MiB.
+let kMSDefaultDatabaseSize = 10 * 1024 * 1024
+
+class AppCenterViewController : NSViewController, NSTextFieldDelegate {
+
+  var appCenter: AppCenterDelegate = AppCenterProvider.shared().appCenter!
+  var currentAction = AuthenticationViewController.AuthAction.signin
+
+  let kMSAppCenterBundleIdentifier = "com.microsoft.appcenter";
+
+  @IBOutlet var installIdLabel : NSTextField?
+  @IBOutlet var appSecretLabel : NSTextField?
+  @IBOutlet var logURLLabel : NSTextField?
+  @IBOutlet var userIdLabel : NSTextField?
+  @IBOutlet var setEnabledButton : NSButton?
+    
+  @IBOutlet weak var deviceIdField: NSTextField!
+  @IBOutlet weak var startupModeField: NSComboBox!
+  @IBOutlet weak var storageMaxSizeField: NSTextField!
+  @IBOutlet weak var storageFileSizeField: NSTextField!
+  @IBOutlet weak var signInButton: NSButton!
+  @IBOutlet weak var signOutButton: NSButton!
+  @IBOutlet weak var overrideCountryCodeButton: NSButton!
+
+  private var dbFileDescriptor: CInt = 0
+  private var dbFileSource: DispatchSourceProtocol?
+
+  deinit {
+      self.dbFileSource?.cancel()
+      close(self.dbFileDescriptor)
+      UserDefaults.standard.removeObserver(self, forKeyPath: kMSStorageMaxSizeKey)
+  }
+
+  override func viewWillAppear() {
+    setEnabledButton?.state = appCenter.isAppCenterEnabled() ? 1 : 0
+  }
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    installIdLabel?.stringValue = appCenter.installId()
+    appSecretLabel?.stringValue = appCenter.appSecret()
+    logURLLabel?.stringValue = appCenter.logUrl()
+    userIdLabel?.stringValue = UserDefaults.standard.string(forKey: kMSUserIdKey) ?? ""
+    setEnabledButton?.state = appCenter.isAppCenterEnabled() ? 1 : 0

```

Comment:
Can we use the enum values supplied by `UIControlState` instead of "magic" numbers `1` and `0`?